test/phpunit/PdfDocTest.php

```diff
@@ -141,13 +141,13 @@ public function testPdfDocGetLineDesc()
 		$db=$this->savdb;
 
 		$localproduct=new Product($this->savdb);
-		$result = $localproduct->fetch(0, 'PIDRESS');
+		$result = $localproduct->fetch(0, 'PRODUCT_SPEC');
 		if ($result < 0) {
-			print "\n".__METHOD__." Failed to make the fetch of product PIDRESS. ".$localproduct->error; die(1);
+			print "\n".__METHOD__." Failed to make the fetch of product PRODUCT_SPEC. ".$localproduct->error; die(1);
 		}
 		$product_id = $localproduct->id;
-		if ($product_id <= 0) {
-			print "\n".__METHOD__." A product with ref PIDRESS must exists into database"; die(1);
+		if ($product_id < 0) {
```

Comment:
Here the goal of test is to ensure we have some data into database that is ready to make the phpunit test.
If product PIDRES does not exists it returns an error.
If you test < 0, you will not have error it it does not exists and the test later with goel to check that the REF or product is correctly inserted into the label is not done.

Here there is no really fix to do except asking the user that run the test to have the correct data ready in its database (or enhance the phpunit test by adding the product before making the test)